examples/global.rs

```diff
@@ -0,0 +1,16 @@
+use clap::Parser;
+
+use once_cell::sync::Lazy;
+
+#[derive(Parser, Debug)]
+#[command(author, version)]
+struct Args {
+    #[arg(short, long, default_value_t = 42)]
+    the_answer: u8,
+}
+
+static ARGS: Lazy<Args> = Lazy::new(Args::parse);
```

Comment:
Could you expand on why you feel this is important enough that we should be advertising it in an example?

I feel like there are enough caveats to this that it will do more harm than good.  For example, see https://github.com/clap-rs/clap/discussions/4234

Reply:
That indeed opinion oriented, while I think library should clearly not use global if really strictly necessary, in application I'm less pedantic. The cost of global like this should be ok for most application.

Global pattern for program options is a very used pattern and I think that clap should advice to use it more specially for beginner. Thus I agree that robust application should pass avoid it.

I think add a simple example of global (maybe I should rename it singleton ?) will help beginner to have a "simple" way to bootstrap their little project without robust need of project like cargo. This example have been inspired by this SO question where OP have trouble with basic, https://stackoverflow.com/q/73860727/7076153.

Reply:
I do not clearly understand the issues of #4234 it's very unclear.